Assets/Scripts/BehaviorTree/Node.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+
+namespace BehaviorTree
+{
+    public enum NodeState
+    {
+        RUNNING,
+        SUCCESS,
+        FAILURE
+    }
+
+    public class Node
+    {
+        protected NodeState State;
+
+        public Node Parent;
+        protected List<Node> Children = new List<Node>();
+
+        private readonly Dictionary<string, object> _dataContext = new Dictionary<string, object>();
+
+        protected Node()
+        {
+            Parent = null;
+        }
+
+        protected Node(List<Node> children)
+        {
+            foreach (var child in children)
+            {
+                Attach(child);
+            }
+        }
+
+        private void Attach(Node node)
```

Comment:
```Attach``` should be exposed to public and return this instance to provide a better chaining API.

Reply:
public + return this + done